### PR TITLE
Typescript translate

### PR DESCRIPTION
**Pull request by ZoicanDenisUi**
Adding typescript

### CODE REVIEW

File: Public/Helpers/mathHelper.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function randomIntFromInterval(min, max) {
```
Comment (anatolie-darii) [inline Public/Helpers/mathHelper.js]: You shouldn't have any .js files in the repo since these will be the compiled files

Comment (ZoicanDenisUi) [reply Public/Helpers/mathHelper.js]: Changed the .gitignore file to ignore those files

**Review by ioanajoj — CHANGES_REQUESTED** [review]
I would clean `tsconfig.json` a bit and remove unnecessary stuff

File: Src/Helpers/domInteractions.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { randomIntFromInterval } from "./mathHelper.js"
```
Comment (anatolie-darii) [inline Src/Helpers/domInteractions.ts:1]: you should import from "./mathHelper" as long as that is a module.

File: Src/HelpingClasses/circle.ts [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
+    
+    step:number
+    direction:Direction
+    slope:number
+    dom:HTMLDivElement
+    circleRadius:number
+    x:number
+    y:number
+
+    constructor(dom:HTMLDivElement,circleRadius:number,width:number,height:number,left:number,top:number){
```
Comment (anatolie-darii) [inline Src/HelpingClasses/circle.ts]: in order to simplify declarations and initializations you can do `constructor(private dom: HTMLDivElement, ....)` and then omit both the declaration for dom and the assignement.
See an explanation here: https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/

File: Src/Helpers/domInteractions.ts (hunk 2 above)
Comment (ZoicanDenisUi) [reply Src/Helpers/domInteractions.ts:1]: I used Webpack to implement this. I created a new PR for this.

File: Src/HelpingClasses/circle.ts (hunk 3 above)
Comment (ZoicanDenisUi) [reply Src/HelpingClasses/circle.ts]: I changed the constructor.

File: .gitignore [hunk 4]
```diff
@@ -1 +1,4 @@
-node_modules/
\ No newline at end of file
+node_modules/
+Public/Helpers/
```
Comment (anatolie-darii) [inline .gitignore]: Ignore the entire Public folder maybe? Also just a nitpick you can name it `dist` or `compiled` or `bin` or something else that is widely used in different frontend frameworks

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "compilerOptions": {
+    "target": "es6",                                
+    "module": "es2015",                          
+    "outDir": "./Public",                       
+    "rootDir": "./Src",
```
Comment (anatolie-darii) [inline tsconfig.json]: nitpick: rename `Src/` to `src` as most naming conventions on the FE are lowercase.

**Review by anatolie-darii — APPROVED** [review]

**Review by ioanajoj — APPROVED** [review]

**Pull request merged**